### PR TITLE
Fix `_hideDetailViewController` not being reset properly

### PR DESCRIPTION
**Pull request by nicoco007**
`_hideDetailViewController` doesn't get reset properly in `LevelCollectionNavigationController.DidActivate` and causes a race condition that doesn't dismiss view controllers properly on the level end screen if:

1. a song is automatically selected by BSL when coming from the main menu
2. the active tab is a non-level-pack tab (i.e. favorites or all songs)
3. a view controller is presented on the level end screen's right pane (e.g. by BeatSaviorData)

This causes `ClearChildViewControllers` to be called too early and eventually `HMUI.Screen.SetRootViewController` gets called twice in a row for the same screen, which in turn causes `TransitionCoroutine` to be cancelled so the previous view controller (i.e. the leaderboard) is never dismissed/disabled.

Basically, this is a fix for this issue:
image

### CODE REVIEW

**Comment by nicoco007** [comment]
i'm apparently too tired to function properly
